src/Symfony/Bundle/TwigBundle/DependencyInjection/Compiler/ExtensionPass.php

```diff
@@ -42,5 +42,16 @@ public function process(ContainerBuilder $container)
         if ($container->has('fragment.handler')) {
             $container->getDefinition('twig.extension.httpkernel')->addTag('twig.extension');
         }
+
+        if ($container->has('assets.packages')) {
+            $container->getDefinition('twig.extension.assets')->addTag('twig.extension');
+        } elseif ($container->has('templating.asset.packages')) {
```

Comment:
the issue with this logic is that it does not allow to have both the new ``asset_path()`` function and the old ``asset()`` function in the same project, forcing an all-or-nothing upgrade. This means that the whole bundle ecosystem has to upgrade at the same time to be able to be used together in a project.

IMO, the old extension should always be registered for BC reasons